### PR TITLE
NAT support for BNW

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

**Comment by Vagabond** [comment]
https://gist.github.com/Vagabond/6c9c1acaf199bd146a35

File: src/riak_repl2_ip.erl [hunk 1]
```diff
@@ -221,15 +216,91 @@ find_best_ip(MyIPs, MyIP, Port, MyCIDR, SearchDepth) ->
...
+            case inet_res:gethostnames(Host) of
+                {ok, {hostent, _Domain, _, inet, 4, Addresses}} ->
+                    lager:debug("remotely resolved ~p to ~p", [Host, Addresses]),
+                    [{External, Addr} || Addr <- Addresses] ++ Acc;
+                Res2 ->
+                    lager:warning("Failed to resolve ~p locally: ~p", [Host, Res]),
+                    lager:warning("Failed to resolve ~p remotely: ~p", [Host, Res2]),
+                    Acc
+            end
+    end;
```
Comment (lordnull) [inline src/riak_repl2_ip.erl:300]: These two clauses duplicate each other quite a bit, seems a good place to abstract some code out to another function.

Comment (Vagabond) [inline src/riak_repl2_ip.erl:300]: In so far as they call similarly named functions with the same return signatures and then do a list comprehension on the result?

Comment (lordnull) [inline src/riak_repl2_ip.erl:300]: The two clauses only differ on the list comprehension, so yes.

**Comment by lordnull** [comment]
Based on the gist, I got a cluster behind nat up and running, and a cluster outside the nat able to talk to it. Correct IP/Ports were retrieved when I configured it right, so this gets a +1 from me. This also tested https://github.com/basho/riak_ee/pull/132 , giving that a +1 as well.

**Comment by bookshelfdave** [comment]
rebased in https://github.com/basho/riak_repl/pull/282

**Pull request closed without merging**